docs/development/go/go-structures/index.md

```diff
@@ -18,235 +19,532 @@ external_resources:
+
+{{< file "employee.go" go >}}
+package main
+
+import (
+    "fmt"
+)
+
+type Employee struct {
+    FirstName string
+    employeeID int
 }
 
-func createStruct(n, s string, h int32) *myStructure {
-    if h > 300 {
-        h = 0
-    }
-    return &myStructure{n, s, h}
+func main() {
+    employee1 := Employee{"Nathan", 8124011}
+    fmt.Println("employee1:", employee1)
+    employee2 := employee1
+    employee2.FirstName = "Andy"
+    employee2.employeeID = 1231410
+    employee1.FirstName = "Nate"
+    fmt.Println("employee1:", employee1)
+    fmt.Println("employee2:", employee2)
 }
 {{< /file >}}
 
-This approach for creating new structure variables allows you to check whether the provided information is correct and valid in advance. Additionally, with this approach you have a central point where structure variables are initialized so if there is something wrong with your structure variables, you know exactly where to look.
+The output of `employee.go` will be:
 
-{{< note >}}
-For those of you with a C or C++ background, it is perfectly legal for a Go function to return the memory address of a local variable. Nothing gets lost so everybody is happy!
-{{< /note >}}
+    go run employee.go
 
-### Example
+{{< output >}}
+employee1: {Nathan 8124011}
+employee1: {Nate 8124011}
+employee2: {Andy 1231410}
+{{< /output >}}
 
-The usage of Go functions that work with structures is illustrated in `functions.go`.
+The `employee2 := employee1` assignment creates a copy of `employee1` and saves it in `employee2`. Changing the `employee1` variable will not affect the contents of `employee2` after the assignment.
 
-{{< file "./functions.go" go >}}
+### Value Semantics with Functions
+
+A struct can be passed to a function. By default, the struct will be copied to its function argument variable. Consider this example:
+
+{{< file "employee.go" go >}}
 package main
 
 import (
     "fmt"
 )
 
-type myStructure struct {
-    Name    string
-    Surname string
-    Height  int32
+type Employee struct {
+    FirstName string
+    employeeID int
 }
 
-func createStruct(n, s string, h int32) *myStructure {
-    if h > 300 {
-        h = 0
-    }
-    return &myStructure{n, s, h}
+func ChangeEmployeeID(e Employee, newID int) {
+    e.employeeID = newID
 }
 
-func retStructure(n, s string, h int32) myStructure {
-    if h > 300 {
-        h = 0
-    }
-    return myStructure{n, s, h}
+func main() {
+    employee1 := Employee{"Nathan", 8124011}
+    fmt.Println(employee1)
+    ChangeEmployeeID(employee1, 1012843)
+    fmt.Println(employee1)
+}
+{{< /file >}}
+
+The output of `employee.go` will be:
+
+    go run employee.go
+
+{{< output >}}
+{Nathan 8124011}
+{Nathan 8124011}
+{{< /output >}}
+
+Calling the `ChangeEmployeeID` function has no effect on the value of `employee` outside of the function scope. As a result, the output of the print statement on line 20 will be the same as the output of line 18's print statement.
+
+## Pointers and Structs
```

Comment:
Top-line changes in this section:

- Continued with the same struct/field naming from earlier to be consistent
- Added details about implicit dereferencing, while still including the syntax for explicit dereferencing
- Added some information about how pointers allow you to mutate structs from multiple variables, in contrast with the earlier value semantics information
- Added a subsection about how passing a pointer to a function allows you to mutate the struct fields from inside the function